From 0a876cb8df14485fc57c435bf7d9554c88b8475e Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Thu, 30 Jun 2011 09:53:05 +0800
Subject: [PATCH 175/233] powerpc/booke: Add Stack Marking support to Booke Exception Prolog

It is from FSL vendor SDK 2.x.

This patch adds a marker to the exception stack frame to aid in debugging.
it's already inserted on other platforms and xmon recognizes it and
identifies exception frames when showing stack traces.

Signed-off-by: Torez Smith  <lnxtorez@linux.vnet.ibm.com>
Signed-off-by: Dave Kleikamp <shaggy@linux.vnet.ibm.com>
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
[context adjustment]
Integrated-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/powerpc/kernel/head_booke.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/head_booke.h b/arch/powerpc/kernel/head_booke.h
index 0273c9c..a1201d9 100644
--- a/arch/powerpc/kernel/head_booke.h
+++ b/arch/powerpc/kernel/head_booke.h
@@ -1,6 +1,8 @@
 #ifndef __HEAD_BOOKE_H__
 #define __HEAD_BOOKE_H__
 
+#include <asm/ptrace.h>        /* for STACK_FRAME_REGS_MARKER */
+
 /*
  * Macros used for common Book-e exception handling
  */
@@ -51,6 +53,9 @@
 	lwz	r13, THREAD_NORMSAVE(2)(r10); /* restore r13 */		     \
 	mflr	r10;							     \
 	stw	r10,_LINK(r11);						     \
+	lis     r10, STACK_FRAME_REGS_MARKER@ha;/* exception frame marker */ \
+	addi    r10, r10, STACK_FRAME_REGS_MARKER@l;                         \
+	stw     r10, 8(r11);                                                 \
 	mfspr	r12,SPRN_SRR0;						     \
 	stw	r1, GPR1(r11);						     \
 	mfspr	r9,SPRN_SRR1;						     \
-- 
1.7.0.4

